build_support/android_builder.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) Intel Corp.  2018.  All Rights Reserved.
+
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+
+# The above copyright notice and this permission notice (including the
+# next paragraph) shall be included in all copies or substantial
+# portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE COPYRIGHT OWNER(S) AND/OR ITS SUPPLIERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#  **********************************************************************/
+#  * Authors:
+#  *   Clayton Craft <clayton.a.craft@intel.com>
+#  **********************************************************************/
+import glob
+import multiprocessing
+import os
+from . import Options
+from . import ProjectMap
+from . import run_batch_command
+from . import rmtree
+from . import Export
+from . import RepoSet
+from . import ProjectInvoke
+from . import Jenkins
+from . import RevisionSpecification
+from . import get_conf_file
+from . import TestLister
+from . import NoConfigFile
+
+
+def cpu_count():
+    cpus = multiprocessing.cpu_count() + 1
+    if cpus > 18:
+        cpus = 18
+    return cpus
+
+
+class AndroidBuilder(object):
+    def __init__(self, extra_definitions=None, src_location="~/android-ia",
+                 module="libGLES_mesa"):
```

Comment:
Since this is supposed to be generic, and consumers are meant to subclass, why not make these arguments positional, and let the subclass add defaults if it wants them?